tiledb/sm/query/reader.cc

```diff
@@ -1294,6 +1313,9 @@ Status Reader::copy_fixed_cells(
   auto it = buffers_.find(name);
   auto buffer_size = it->second.buffer_size_;
   auto cell_size = array_schema_->cell_size(name);
+  if (query_datatypes_.find(name) != query_datatypes_.end()) {
+    cell_size = (uint64_t)(cell_size * calc_query_size_ratio(name));
```

Comment:
assign `cell_size` directly to the value in `query_datatypes_` rather than scaling it by a ratio.

Reply:
yes, will do that